### PR TITLE
Toplevel waf powershell changes

### PR DESCRIPTION
**Pull request by venk8**
## Checklist

- [ ] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/AzureApplicationGatewayFirewallPolicyManagedRuleSet.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            base.ExecuteCmdlet();
+        }
+
+        protected PSApplicationGatewayFirewallPolicyManagedRuleSet NewObject()
+        {
+            return new PSApplicationGatewayFirewallPolicyManagedRuleSet()
+            {
+                RuleSetType = this.RuleSetType,
+                RuleSetVersion = this.RuleSetVersion,               
+                RuleGroupOverrides = this.RuleGroupOverrides?.ToList()
```
Comment (number213) [inline src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/AzureApplicationGatewayFirewallPolicyManagedRuleSet.cs]:
```suggestion
                RuleGroupOverrides = this.RuleGroupOverride?.ToList()
```

File: src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/ManagedRuleGroupOverride/AzureApplicationGatewayFirewallPolicyManagedRuleGroupOverride.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            base.ExecuteCmdlet();
+        }
+
+        protected PSApplicationGatewayFirewallPolicyManagedRuleGroupOverride NewObject()
+        {
+            return new PSApplicationGatewayFirewallPolicyManagedRuleGroupOverride()
+            {
+                RuleGroupName = this.RuleGroupName,
+                Rules = this.Rules?.ToList()
```
Comment (number213) [inline src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/ManagedRuleGroupOverride/AzureApplicationGatewayFirewallPolicyManagedRuleGroupOverride.cs]:
```suggestion
                Rules = this.Rule?.ToList()
```

File: src/Network/Network/FirewallPolicy/NewAzureApplicationGatewayFirewallPolicyCommand.cs [hunk 3]
```diff
@@ -96,6 +108,8 @@ private PSApplicationGatewayWebApplicationFirewallPolicy CreateApplicationGatewa
             firewallPolicy.ResourceGroupName = this.ResourceGroupName;
             firewallPolicy.Location = this.Location;
             firewallPolicy.CustomRules = this.CustomRule?.ToList();
+            firewallPolicy.ManagedRules = this.ManagedRules;
```
Comment (number213) [inline src/Network/Network/FirewallPolicy/NewAzureApplicationGatewayFirewallPolicyCommand.cs]:
```suggestion
            firewallPolicy.ManagedRules = this.ManagedRule;
```

File: src/Network/Network/FirewallPolicy/NewAzureApplicationGatewayFirewallPolicyCommand.cs [hunk 4]
```diff
@@ -53,11 +53,23 @@ public class NewAzureApplicationGatewayFirewallPolicyCommand : ApplicationGatewa
...
+         Mandatory = false,
+         ValueFromPipelineByPropertyName = true,
+         HelpMessage = "The list of CustomRules")]
         public PSApplicationGatewayFirewallCustomRule[] CustomRule { get; set; }
 
+        [Parameter(
+         Mandatory = false,
+         ValueFromPipelineByPropertyName = true,
+         HelpMessage = "Policysettings of the firewall policy")]
+        public PSApplicationGatewayFirewallPolicySettings PolicySettings { get; set; }
```
Comment (venkatsvpr) [inline src/Network/Network/FirewallPolicy/NewAzureApplicationGatewayFirewallPolicyCommand.cs]: @markcowl Can this be PolicySettings.. or should I change it as PolicySetting?

**Comment by venkatsvpr** [comment]
@markcowl  I have addressed the review comments that were given by @number213. Please take a look and let me know if some more changes are required.

**Comment by markcowl** [comment]
cmdlet review is here: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/393

Note that this PR did not have cmdlet review started until 4 days after review complete and  will *not* make it into the 10/15 release, you will need to retarget it.

**Comment by number213** [comment]
Closing until branch `network-october` is created for the next release

**Comment by number213** [comment]
Retargeted to `network-october`. @venkatsvpr please pull the latest from upstream

File: src/Network/Network/help/New-AzApplicationGatewayFirewallPolicyExclusion.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Network.dll-Help.xml
+Module Name: Az.Network
+online version: https://docs.microsoft.com/en-us/powershell/module/az.network/new-azapplicationgatewayfirewallexclusionconfig
```
Comment (number213) [inline src/Network/Network/help/New-AzApplicationGatewayFirewallPolicyExclusion.md]: URI is incorrect, there is no "config" at the end of cmdlet name

File: src/Network/Network/help/New-AzApplicationGatewayFirewallPolicyManagedRuleGroupOverride.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Network.dll-Help.xml
+Module Name: Az.Network
+online version: https://docs.microsoft.com/en-us/powershell/module/az.network/new-azapplicationgatewayfirewallcondition
```
Comment (number213) [inline src/Network/Network/help/New-AzApplicationGatewayFirewallPolicyManagedRuleGroupOverride.md]: Incorrect URI, references different cmdlet

Comment (venkatsvpr) [reply src/Network/Network/help/New-AzApplicationGatewayFirewallPolicyManagedRuleGroupOverride.md]: Fixed.

File: src/Network/Network/help/New-AzApplicationGatewayFirewallPolicyExclusion.md (hunk 5 above)
Comment (venkatsvpr) [reply src/Network/Network/help/New-AzApplicationGatewayFirewallPolicyExclusion.md]: Fixed

**Comment by msJinLei** [comment]
@number213 could you fix the markdown file generation error?

**Comment by number213** [comment]
We've checked in test recordings with API version 2019-09-01 into `network-october`, please merge/rebase to unblock your PR

**Comment by venkatsvpr** [comment]
@msJinLei Please review the changes. If they are fine. I will merge the pull request.

File: src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/ManagedRuleGroupOverride/AzureApplicationGatewayFirewallPolicyManagedRuleGroupOverride.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace Microsoft.Azure.Commands.Network
+{
+    public class AzureApplicationGatewayFirewallPolicyManagedRuleGroupOverride : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Rule Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string RuleGroupName { get; set; }
```
Comment (msJinLei) [inline src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/ManagedRuleGroupOverride/AzureApplicationGatewayFirewallPolicyManagedRuleGroupOverride.cs]: Any arugument completer for rulegroupname?

File: src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/ManagedRuleGroupOverride/ManagedRuleOverride/AzureApplicationGatewayFirewallPolicyManagedRuleOverride.cs [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+            HelpMessage = "Rule Id.")]
+        [ValidateNotNullOrEmpty]
+        public string RuleId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "State of the Rule.")]
+        [ValidateSet("Disabled", IgnoreCase = true)]
+        [ValidateNotNullOrEmpty]
+        public string State { get; set; }
```
Comment (msJinLei) [inline src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/ManagedRuleGroupOverride/ManagedRuleOverride/AzureApplicationGatewayFirewallPolicyManagedRuleOverride.cs]: Will have enabled value later? if not, why not use switch parameter

Comment (venkatsvpr) [reply src/Network/Network/FirewallPolicy/ManagedRules/ManagedRuleSet/ManagedRuleGroupOverride/ManagedRuleOverride/AzureApplicationGatewayFirewallPolicyManagedRuleOverride.cs]: Yes. We are planning to introduce the Enabled value later.

**Comment by number213** [comment]
This PR depends on https://github.com/Azure/azure-powershell/pull/10386

**Comment by number213** [comment]
@msJinLei your comments have been addressed, please take a look

**Pull request merged**